File: array/1089_duplicate_zeros/1089_duplicate_zeros.py
from typing import List


class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        nums = []
        for num in arr:
            nums.append(num)
            if num == 0:
                nums.append(0)
        for i in range(len(arr)):
            arr[i] = nums[i]


arr = [1,0,2,3,0,4,5,0]
Solution().duplicateZeros(arr)
print(arr)

File: array/1089_duplicate_zeros/1089_duplicate_zeros.go
package main

import "fmt"

func duplicateZeros(arr []int) {
	for index := 0; index < len(arr); index++ {
		if arr[index] == 0 && index+1 < len(arr) {
			arr = append(arr[:index+1], arr[index:len(arr)-1]...)
			index++
		}
	}
}

func main() {
	arr := []int{1, 0, 2, 3, 0, 4, 5, 0}
	duplicateZeros(arr)
	fmt.Println(arr)
}
